network/netcore/src/transport/tcp.rs

```diff
@@ -75,27 +75,27 @@ impl Transport for TcpTransport {
     ) -> Result<(Self::Listener, NetworkAddress), Self::Error> {
         let ((ipaddr, port), addr_suffix) =
             parse_ip_tcp(addr.as_slice()).ok_or_else(|| invalid_addr_error(&addr))?;
+        if !addr_suffix.is_empty() {
+            return Err(invalid_addr_error(&addr));
+        }
 
         let listener = ::std::net::TcpListener::bind((ipaddr, port))?;
         let listener = TcpListener::try_from(listener)?;
-
-        // append the addr_suffix so any trailing protocols get included in the
-        // actual listening adddress we return
-        let actual_addr =
-            NetworkAddress::from(listener.local_addr()?).extend_from_slice(addr_suffix);
+        let listen_addr = NetworkAddress::from(listener.local_addr()?);
 
         Ok((
             TcpListenerStream {
                 inner: listener,
                 config: self.clone(),
             },
-            actual_addr,
+            listen_addr,
         ))
     }
 
     fn dial(&self, addr: NetworkAddress) -> Result<Self::Outbound, Self::Error> {
         let (addr_string, _addr_suffix) =
             parse_addr_and_port(addr.as_slice()).ok_or_else(|| invalid_addr_error(&addr))?;
```

Comment:
this should also come from NetworkAddress directly and in that way we can enforce your todo.... since it should ideally return a sock addr with a resolved dns...

Reply:
I don't think `parse_addr_and_port` belongs in `NetworkAddress`. it's just a temporary, minimum effort way to get dialing working. doing the proper filtering requires a bit more work.

Reply:
why's that? what's the plan?

Reply:
because the end state would probably look like

```rust
parse_ip_tcp => tokio::net::ToSocketAddrs => .await  \
                                                      => maybe filter ipv4/ipv6 => try SocketAddrs until one works
parse_dns_tcp => tokio::net::ToSocketAddrs => .await /
```

like this, but filtering ipv4/ipv6
https://docs.rs/tokio/0.2.21/src/tokio/net/tcp/stream.rs.html#97-115

hence parse_addr_and_port is temporary and i want to remove it soon.